app/client/src/utils/helpers.tsx

```diff
@@ -346,3 +347,14 @@ export const getCanManage = (currentOrg: Org) => {
   );
   return canManage;
 };
+
+export const getIsSafeRedirectURL = (redirectURL: string) => {
+  try {
+    const currentURL = window.location.href;
+    const currentHost = new URL(currentURL).hostname;
+    const redirectURLHost = new URL(redirectURL).hostname;
+    return getIsSafeURL(redirectURL) && currentHost === redirectURLHost;
```

Comment:
The `getIsSafeURL` is still too permissive in my opinion. It allows many more schemes in addition to `http` or `https`. I wouldn't say we change the implementation of `getIsSafeURL`, such a function might be useful elsewhere as well, but in this case, something like this might be better and simpler:

```suggestion
    return new URL(redirectURL).origin === window.location.origin;
```

The `.origin` field for `http` or `https` URLs returns a string with scheme, hostname and port. I'd say for a URL that is allowed as a redirect URL, _all_ these three should match. I don't want to allow `tel:` and `mailto:` URLs. We don't need them here, and the just increase the vulnerability surface area. What say?

Reference docs:

- <https://developer.mozilla.org/en-US/docs/Web/API/Location/origin>
- <https://developer.mozilla.org/en-US/docs/Web/API/URL/origin>

Reply:
Yeah sounds good, this check should be enough